beep/utils/waveform.py

```diff
@@ -56,3 +58,246 @@ def convert_velocity_to_power_waveform(waveform_file, velocity_units):
     power = -power * df.v  # positive power = charge
 
     return pd.DataFrame({"time": df.t, "power": power})
+
+
+class RapidChargeWave:
+    """
+       Object to produce charging waveforms, with direct comparison between stepwise and
+       smooth charging waveforms. Algorithms written by Patrick Asinger.
+
+    """
+    def __init__(self,
+                 final_c_rate,
+                 soc_initial,
+                 soc_final,
+                 max_c_rate,
+                 min_c_rate
+                 ):
+        """
+        Args:
+            final_c_rate (float): charging rate for the final step
+            soc_initial (float): estimated starting soc for the fast charging portion of the cycle
+            soc_final (float): estimated soc to end the fast charging portion of the cycle
+            max_c_rate (float): maximum charging rate for the charge
+            min_c_rate (float): minimum charging rate for the charge
+
+        """
+        self.final_c_rate = final_c_rate
+        self.soc_i = soc_initial
+        self.soc_f = soc_final
+        self.max_c_rate = max_c_rate
+        self.min_c_rate = min_c_rate
+        self.soc_points = 1000
```

Comment:
Can we make `soc_points` a global/config variable (if it is not going to be passed as an argument to the constructor)?

Reply:
I don't think that making it a global variable is a good idea here since we might need to alter the interpolation. Is there a specific reason you were thinking it should be a global variable?

Reply:
No strong reason -- I wasn't sure if having a hardcoded variable buried in was better than having it defined as a config/global variable when we needed to change. But, since its a class variable, I suppose its ok!